### PR TITLE
Version Bumping

### PR DESCRIPTION
**Pull request by tduffield**
* Use Ubuntu 14.04 as default in omnibus-dev build
* Update Ruby deps to match Chef 12.14.13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -51,8 +51,8 @@ install:
   - luarocks install --local lpeg
   - luarocks install --local lua-cjson
   - eval $(luarocks path)
-  - rvm install --binary 2.1.4
-  - rvm use 2.1.4
+  - rvm install --binary 2.3.1
```
Comment (marcparadise) [inline .travis.yml]: Do we need to update the `CHEFDK_GECODE_PATH` for the new version number?  (L39)

**Comment by marcparadise** [comment]
:+1: save the one question, we'll see how travis & pipeline like it...

File: .travis.yml (hunk 1 above)
Comment (tduffield) [inline .travis.yml]: ChefDK on master is still using Ruby 2.1

File: omnibus/Gemfile.lock [hunk 2]
```diff
@@ -1,19 +1,22 @@
 GIT
   remote: git://github.com/chef/omnibus-software.git
-  revision: 40fbb9d412a2ebc17c06ff34922682de28970127
+  revision: b1a068c75d4eb0833c6c3884f7e0692ab1f29fb1
+  branch: ksubrama/ruby23
```
Comment (stevendanna) [inline omnibus/Gemfile.lock]: I'm really not thrilled with the number of projects where we are now on this branch.  We should help get it merged if possible.

File: src/opscode-expander/Gemfile [hunk 3]
```diff
@@ -2,7 +2,7 @@
 source "http://rubygems.org"
 
 gem "amqp", "~> 0.6.7"
-gem "eventmachine", '~> 0.12.10'
+gem "eventmachine", '~> 1.2.0'
```
Comment (stevendanna) [inline src/opscode-expander/Gemfile]: This is a big jump.  I think we have to do it eventually, but we'll want to keep it in mind if there are expander issues in the near future.

File: omnibus/Gemfile.lock (hunk 2 above)
Comment (tduffield) [inline omnibus/Gemfile.lock]: For sure! I'm going to keep in touch with maintainers :)

**Comment by stevendanna** [comment]
Looks reasonable

File: src/opscode-expander/Gemfile (hunk 3 above)
Comment (tduffield) [inline src/opscode-expander/Gemfile]: Yep, I pulled this in from @sersut's #907 change.

**Pull request merged**